app/models/name.rb

```diff
@@ -1639,22 +1679,33 @@ def self.parse_author(str)
   end
 
   def self.parse_group(str, deprecated = false)
-    results = nil
-    if match = GROUP_PAT.match(str)
-      name = match[1]
-      text_name = name.tr("ë", "e")
-      parent_name = name.sub(LAST_PART, "")
-      results = ParsedName.new(
-        text_name: text_name + " group",
-        search_name: text_name + " group",
-        sort_name: format_sort_name(text_name, "group"),
-        display_name: format_name(name, deprecated) + " group",
-        parent_name: parent_name,
-        rank: :Group,
-        author: ""
-      )
+    return unless match = GROUP_PAT.match(str)
+
+    # Parse the string without "group"
+    ungrouped_str = str.sub(/\s#{GROUP_ABBR}/, "")
+    result = parse_name(ungrouped_str)
+    return nil unless result
+
+    # Adjust the parsed name
+    result.text_name += " group"
+
+    if result.author.present?
+      # Add "group" before author
+      author = Regexp.escape(result.author)
+      result.search_name.sub!( /(.*)(#{author})/, '\1group \2')
+      result.sort_name.sub!(   /(.*)(#{author})/, '\1 group  \2')
+      result.display_name.sub!(/(.*)(#{author})/, '\1group \2')
+    else
```

Comment:
Another great catch. I think we should make that assumption.  (And I had already planned to tweak the instructions in the create_name template under Scientific Name to tell users where to put "group" or "clade"):
>Unformatted taxon name <i>without</i> author info or status info. E.g.: <br>&nbsp;&nbsp; Amanita muscaria var. flavivolvata <br>For a <b>group or clade</b>, select <b>[:RANK]</b> “[:RANK_GROUP]” above, and add either “group” or “clade” at the end of <b>[:form_names_text_name]</b>. E.g.: <br>&nbsp;&nbsp; Amanita muscaria group <br>&nbsp;&nbsp; Morchella elata clade.

(I can only think of one case where Author doesn't legitimately come at the end: autonyms like _Amanita_ Pers. sect. _Amanita_. But "group" will blow up that case in any event.
And, of course, there will always be a few Names where users ignore the instructions and try to do things like Amanita crocea sensu auct. amer. group.)